app/helpers/challenge_helper.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+# ChallengeHelper is the helper for challenge show page in public facing pages.
+module ChallengeHelper
+
+  # submission period info for challenge
+  def format_submission_period(challenge)
+    phases = challenge.phases
+
+    if single_phase?(challenge)
+      {
+        phase: phases.first,
+        status_text: submission_status_text(phases.first),
+        closing_soon: phase_closing_soon?(phases.first)
+      }
+    else
+      current_phase = get_current_phase(phases)
+      previous_phase = get_previous_phase(phases)
+      next_phase = get_next_phase(phases)
+      {
+        status_text: multi_phase_status_text(challenge, current_phase, previous_phase, next_phase)
+      }
+    end
+  end
+
+  # challenge type(s)
+  def types_text(challenge)
+    if challenge.types.any?
+      "#{challenge.primary_type}; #{challenge.types.join('; ')}"
+    else
+      challenge.primary_type
+    end
+  end
+
+  def follow_button_text(challenge)
+    if challenge.gov_delivery_subscribers > 0
+      "Follow challenge (#{challenge.gov_delivery_subscribers})"
+    else
+      "Follow challenge"
+    end
+  end
+
+  # challenge timeline
+  def format_datetime(datetime)
+    datetime.strftime("%B %d, %Y %I:%M %p")
+  end
```

Comment:
do we need both this _and_ `format_local_datetime`? whichever one we use, can you add the timezone %Z?
```suggestion
  def format_datetime(datetime)
    datetime.strftime("%B %d, %Y %I:%M %p %Z")
  end
```